File: mcorg.backends/permission-service/src/main/java/app/mcorg/permission/infrastructure/service/entities/XstsRequest.java
package app.mcorg.permission.infrastructure.service.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public record XstsRequest(@JsonProperty("Properties") Properties properties,
                          @JsonProperty("RelyingParty") String relyingParty,
                          @JsonProperty("TokenType") String tokenType) {

    public static XstsRequest create(String xboxToken) {
        return new XstsRequest(
                new Properties("RETAIL", new String[]{xboxToken}),
                "rp://api.minecraftservices.com/",
                "JWT"
        );
    }

    private record Properties(@JsonProperty("SandboxId") String sandboxId,
                              @JsonProperty("UserTokens") String[] userTokens
    ) {
    }
}

File: webapp/src/main/kotlin/app/mcorg/pipeline/auth/domain/XstsRequest.kt
package app.mcorg.pipeline.auth.domain

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class XstsRequest(@SerialName("Properties") val properties: XstsProperties,
                       @SerialName("RelyingParty") val relyingParty: String,
                       @SerialName("TokenType") val tokenType: String)